arch/arm/include/cortex_m/cmse.h

```diff
@@ -0,0 +1,223 @@
+ *                    mode is privileged or unprivileged.
+ *
+ * @return 1 if address is readable, 0 otherwise.
+ */
+int arm_cmse_addr_read_ok(u32_t addr, int force_npriv);
+
+/**
+ * @brief Read and Write accessibility of an address
+ *
+ * Evaluates whether a specified memory location can be read/written according
+ * to the permissions of the current state MPU and the specified operation
+ * mode.
+ *
+ * This function shall always return zero:
+ * - if executed from an unprivileged mode,
+ * - if the address matches multiple MPU regions.
+ *
+ * @param addr The address for which the RW ability is requested
+ * @param force_npriv Instruct to return the RW ability of the address
+ *                    for unprivileged access, regardless of whether the current
+ *                    mode is privileged or unprivileged.
+ *
+ * @return 1 if address is Read and Writable, 0 otherwise.
+ */
+int arm_cmse_addr_readwrite_ok(u32_t addr, int force_npriv);
+
+/**
+ * @brief Read accessibility of an address range
+ *
+ * Evaluates whether a memory address range, specified by its base address
+ * and size, can be read according to the permissions of the current state MPU
+ * and the specified operation mode.
+ *
+ * This function shall always return zero:
+ * - if executed from an unprivileged mode,
+ * - if the address range overlaps with multiple MPU (and/or SAU/IDAU) regions.
+ *
+ * @param addr The base address of an address range,
+ *             for which the readability is requested
+ * @param size The size of the address range
+ * @param force_npriv Instruct to return the readability of the address range
+ *                    for unprivileged access, regardless of whether the current
+ *                    mode is privileged or unprivileged.
+ *
+ * @return 1 if address range is readable, 0 otherwise.
+ */
+int arm_cmse_addr_range_read_ok(u32_t addr, u32_t size, int force_npriv);
+
+/**
+ * @brief Read and Write accessibility of an address range
+ *
+ * Evaluates whether a memory address range, specified by its base address
+ * and size, can be read/written according to the permissions of the current
+ * state MPU and the specified operation mode.
+ *
+ * This function shall always return zero:
+ * - if executed from an unprivileged mode,
+ * - if the address range overlaps with multiple MPU (and/or SAU/IDAU) regions.
+ *
+ * @param addr The base address of an address range,
+ *             for which the RW ability is requested
+ * @param size The size of the address range
+ * @param force_npriv Instruct to return the RW ability of the address range
+ *                    for unprivileged access, regardless of whether the current
+ *                    mode is privileged or unprivileged.
+ *
+ * @return 1 if address range is Read and Writable, 0 otherwise.
+ */
+int arm_cmse_addr_range_readwrite_ok(u32_t addr, u32_t size, int force_npriv);
+
+/* Required for C99 compilation */
+#ifndef typepof
```

Comment:
This looks weird, is this all correct?

Reply:
Yes, this is needed for the <arm_cmse.h> macro definitions, such as:

cmse_check_pointed_object(p, f)

Reply:
Should this be `#ifndef typeof`

Reply:
Ah, right, there is an ugly typo there. Will fix, thanks